rpmio/rpmkeyring.c

```diff
@@ -276,7 +276,18 @@ rpmRC rpmKeyringVerifySig(rpmKeyring keyring, pgpDigParams sig, DIGEST_CTX ctx)
 	    pgpkey = key->pgpkey;
 
 	/* We call verify even if key not found for a signature sanity check */
-	rc = pgpVerifySignature(pgpkey, sig, ctx);
+        char *errormsg = NULL;
+	rc = pgpVerifySignature2(pgpkey, sig, ctx, &errormsg);
+        if (errormsg) {
+            if (rc) {
+                rpmlog(RPMLOG_ERR, "Error verifying signature: %s",
+                       errormsg);
+            } else {
+                rpmlog(RPMLOG_WARNING, "Warning verifying signature: %s",
```

Comment:
rpmlog() prefixes ERR and WARNING with `error:` and `warning:` respectively so these will appear a bit redundant, eg `error: Error verifying signature: <explanation>`

Not that I have any particularly bright ideas for better messages.

Reply:
Good point, I'll remove the error and warning.  `Verifying signature` is enough.  errormsg will contain the actual text.

Reply:
Having now tested this, I wonder if even that is necessary because that context is already supplied by the Sequoia message:

> /home/pmatilai/Downloads/anydesk-6.1.1-1.el8.x86_64.rpm:
error: Error verifying signature: Verifying a signature using certificate D56311E5FF3B6F39D5A16ABE18DF3741CDFFDE29:
  No binding signature at time 2021-04-13T11:08:37Z
  error: Error verifying signature: Verifying a signature using certificate D56311E5FF3B6F39D5A16ABE18DF3741CDFFDE29: